### ISSUE TITLE
Problems with YOLO fine tunning on COCO keypoints annotation

### ISSUE THREAD

**Issue by JohnConnor123** [ultralytics/ultralytics#18599]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

I bought annotation of key points of the lower body (up to the hip inclusive) and I was sent an archive in the "CVAT for Image" format. I converted a single .xml annotation file into N files for each of the images, but I do not understand what format the annotation file should be in. It requires 7 numbers from me, but I only have 3 (class label and x, y coordinates). I want to get a pose estimation with edges between the points, but after fine tunning I get just class points without connections and with very small confidence scores (I attached an example with a piece of the image). I understand that YOLO perceives the annotation to the image as object detection with very small classes, and not as a pose estimation, but how can I fix this? What format should I provide in the annotation file? Here is an example of an annotation file for a random `cvat_pos_detection/labels/train/0_IMG47.txt` image that looks like this:
```
5 0.294300 0.463956 0.010000 0.010000 0.294300 0.463956
2 0.667033 0.463956 0.010000 0.010000 0.667033 0.463956
4 0.249333 0.117103 0.010000 0.010000 0.249333 0.117103
1 0.724867 0.108100 0.010000 0.010000 0.724867 0.108100
3 0.368200 0.608100 0.010000 0.010000 0.368200 0.608100
0 0.657400 0.720717 0.010000 0.010000 0.657400 0.720717
```
![image](https://github.com/user-attachments/assets/edb2ec53-6883-420a-87cc-de62a206d583)

**Comment by UltralyticsAssistant** [comment]
👋 Hello @JohnConnor123, thank you for your interest in Ultralytics 🚀! We recommend checking out our [Docs](https://docs.ultralytics.com) where you can find comprehensive guidance on various tasks including pose estimation. You may also explore [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples to familiarize yourself with our workflows.

### Addressing Annotation and Pose Estimation
If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) including:
- A clear description of the issue
- Steps to reproduce it
- A subset of your annotation files and training commands/logs

For your specific ❓ question on pose estimation and annotation formats, please make sure you're structuring your data correctly. YOLO format for pose estimation requires each labeled keypoint, along with connections, to conform to specific annotation standards. You can delve into the [custom training guide](https://docs.ultralytics.com/guides/model-training-tips/) for more insights on best practices, including dataset preparation and proper annotation formats.

Additionally, your example annotations and results suggest there might be confusion in the format being fed into the model. Reviewing the expected pose estimation data structure and ensuring that your model is fine-tuned appropriately can help resolve the confidence score and edge connectivity issues you're encountering.

### How to Proceed
1. Ensure you're using the latest version of Ultralytics software. Update via pip:
   ```bash
   pip install -U ultralytics
   ```
2. Use our verified environments for reproducibility:
   - Notebooks with free GPUs: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
   - Set up environments like [Docker](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

### Engage the Community
For further assistance, feel free to join our vibrant community. You can connect in real-time on [Discord](https://discord.com/invite/ultralytics) 🎧, post to [Discourse](https://community.ultralytics.com), or share challenges and insights on our [Subreddit](https://reddit.com/r/Ultralytics).

### Status
You can check the status of our CI workflows here: <a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>. If this badge is green, our tests are verified and running successfully across multiple platforms.

This is an automated response 🤖, and an Ultralytics engineer will assist you further shortly. If you have additional context, feel free to add it here to help us support you better! 🚀

**Comment by visionNoob** [comment]
Here is document for pose dataset: [Pose Estimation Datasets Overview](https://docs.ultralytics.com/datasets/pose/#dataset-yaml-format). 

As i know, for training a `YOLO-pose`,  the annotations must not only include the `keypoint coordinates` but also the `bounding box information for the object itself`. The bounding box provides the location and size of the object (person), acting as a reference  for the keypoints. 

For that reason, annotation looks like
```
<class-index> <x> <y> <width> <height> <px1> <py1> <px2> <py2> ... <pxn> <pyn>
```

**Comment by JohnConnor123** [comment]
> Here is document for pose dataset: [Pose Estimation Datasets Overview](https://docs.ultralytics.com/datasets/pose/#dataset-yaml-format).
> 
> As i know, for training a `YOLO-pose`, the annotations must not only include the `keypoint coordinates` but also the `bounding box information for the object itself`. The bounding box provides the location and size of the object (person), acting as a reference for the keypoints.
> 
> For that reason, annotation looks like
> 
> ```
> <class-index> <x> <y> <width> <height> <px1> <py1> <px2> <py2> ... <pxn> <pyn>
> ```

How to convert keypoints dataset with "CVAT for Image" format to YOLO fine tunning?

**Comment by visionNoob** [comment]
@JohnConnor123 
Do your dataset have bounding box for each object?

**Comment by JohnConnor123** [comment]
> @JohnConnor123 
> Do your dataset have bounding box for each object?

no, but even if there were, I don’t understand where there would be n pairs of coordinates instead of two x,y pairs.

**Comment by JohnConnor123** [comment]
> @JohnConnor123 Do your dataset have bounding box for each object?

Btw, I can get bboxes for a person by automatically running the video through a pre-trained yolo. Because the person takes up about 80% of the photo, there will be no problems with detection. So we will assume that I have the coordinates of the bboxes

**Comment by glenn-jocher** [comment]
If you have bounding boxes, you can integrate them with your keypoint annotations to convert your dataset into the required YOLO pose format. Each line in the annotation file should include the class index, bounding box coordinates (x, y, width, height), and normalized keypoint coordinates. Ensure that your bounding boxes and keypoints are correctly aligned. For further details, you can refer to the [Pose Estimation Dataset Format](https://docs.ultralytics.com/datasets/pose/#dataset-yaml-format).

**Comment by JohnConnor123** [comment]
> If you have bounding boxes, you can integrate them with your keypoint annotations to convert your dataset into the required YOLO pose format. Each line in the annotation file should include the class index, bounding box coordinates (x, y, width, height), and normalized keypoint coordinates. Ensure that your bounding boxes and keypoints are correctly aligned. For further details, you can refer to the [Pose Estimation Dataset Format](https://docs.ultralytics.com/datasets/pose/#dataset-yaml-format).

And what about data.yaml format for this task?

**Comment by JohnConnor123** [comment]
> If you have bounding boxes, you can integrate them with your keypoint annotations to convert your dataset into the required YOLO pose format. Each line in the annotation file should include the class index, bounding box coordinates (x, y, width, height), and normalized keypoint coordinates. Ensure that your bounding boxes and keypoints are correctly aligned. For further details, you can refer to the [Pose Estimation Dataset Format](https://docs.ultralytics.com/datasets/pose/#dataset-yaml-format).

Should bbox coordinates (x, y, w, h) be normalized? Without it I got very bad results, but with it it was better.

**Comment by JohnConnor123** [comment]
> If you have bounding boxes, you can integrate them with your keypoint annotations to convert your dataset into the required YOLO pose format. Each line in the annotation file should include the class index, bounding box coordinates (x, y, width, height), and normalized keypoint coordinates. Ensure that your bounding boxes and keypoints are correctly aligned. For further details, you can refer to the [Pose Estimation Dataset Format](https://docs.ultralytics.com/datasets/pose/#dataset-yaml-format).

And I have 2 more questions:
1. Is it normal that the quality of the model drops after fine-tuning on the dataset?
P.s. Here is some additional information that might help:
Total images: 232
Training images: 185
Validation images: 47
Used YOLO model for fine-tunning: `yolo11s-pose.pt`

2. Is it possible to turn keypoints from inference into a skeleton (connect the dots)? There are no edges on the inference of the fine-tuned model. Maybe I did something wrong (since there were edges in the original) and because of this the quality of the model drops

**Comment by glenn-jocher** [comment]
1. A drop in model performance after fine-tuning can occur due to limited dataset size or distributional differences between your dataset and the pre-trained one (e.g., COCO). You might consider increasing your dataset size, ensuring annotations are accurate, or using techniques like data augmentation to improve results.

2. YOLO does not inherently connect keypoints into a skeleton. Generating skeletons requires defining edges between keypoints post-inference. You can achieve this by mapping keypoints into a skeletal structure using a custom script. The original model might have used visualization tools or additional code for displaying connections, which your current setup may lack.

For further improvement, verify that your dataset format strictly follows the [Pose Estimation Dataset Format](https://docs.ultralytics.com/datasets/pose/#dataset-yaml-format).
